collie/interactions/datasets.py

```diff
@@ -333,45 +342,70 @@ def __init__(self,
             )
 
         # When an ``Interactions`` is instantiated with exact negative sampling, a
-        # ``positive_items`` attribute is created, a ``set`` of the ``mat`` object that enables
-        # fast, O(1), ``(row, col)`` lookup. When ``__getitem__`` is called, negative item IDs are
-        # sampled one-at-a-time from all possible values in ``self.num_items``, we check if that
-        # user ID, item ID pair is in ``self.positive_items``, and sample continuously until we
-        # have a negative match or reach a limit of ``max_number_of_samples_to_consider`` tries
+        # ``positive_sets`` attribute is created, a ``set`` of the ``mat`` object that enables
+        # fast, O(1), ``(row, col)`` lookup. When ``__getitem__`` is called, negative item or
+        # user IDs are sampled one-at-a-time from all possible values in ``self.num_items``
+        # or ``self.num_users``, we check if that user ID, item ID pair is in
+        # ``self.positive_sets``, and sample continuously until we have a negative match or
+        # reach a limit of ``max_number_of_samples_to_consider`` tries
         if self.check_num_negative_samples_is_valid:
             print('Checking ``num_negative_samples`` is valid...')
+            if self.negative_sample_type == 'item':
+                if self._checked_num_negative_item_samples_is_valid is False:
+                    self._checked_num_negative_item_samples_is_valid = (
+                        self._check_num_negative_samples_is_valid()
+                    )
```

Comment:
Since `_check_num_negative_samples_is_valid` will either return `True` or raise an error, I think it's okay to write this out as: 

```suggestion
                    self._check_num_negative_samples_is_valid()
                    self._checked_num_negative_item_samples_is_valid = True
```